contracts/service-registry/src/contract.rs

```diff
@@ -257,6 +258,12 @@ pub mod execute {
 
         for chain in chains {
             WORKERS_PER_CHAIN.save(deps.storage, (&service_name, &chain, &info.sender), &())?;
+
+            let existing_chains =
+                CHAINS_PER_WORKER.may_load(deps.storage, (&service_name, &info.sender))?;
+            let mut chains_names = existing_chains.unwrap_or_else(HashSet::new);
+            chains_names.insert(chain.clone());
+            CHAINS_PER_WORKER.save(deps.storage, (&service_name, &info.sender), &(chains_names))?;
         }
```

Comment:
move this whole part into state so there is no danger of accidentally updating only one of the storage maps?

Reply:
I think doing this would interfere with the new optimized code. As the `may_load` and `save` parts are now out of the loop and the `insert` section is still inside the loop. So moving all these to another function adds overhead I think.

Reply:
@cjcobb23 any comments?